containers-tests/tests/map-properties.hs

```diff
@@ -1365,46 +1450,116 @@ prop_splitModel n ys = length ys > 0 ==>
   in  toAscList l == sort [(k, v) | (k,v) <- xs, k < n] &&
       toAscList r == sort [(k, v) | (k,v) <- xs, k > n]
 
-prop_foldr :: Int -> [(Int, Int)] -> Property
-prop_foldr n ys = length ys > 0 ==>
-  let xs = List.nubBy ((==) `on` fst) ys
-      m  = fromList xs
+prop_fold :: Map Int Int -> Property
+prop_fold m = size m > 0 ==>
+  let xs = toList m
+  in  Foldable.fold (map (:[]) m) == List.map snd xs
+
+prop_foldMap :: Map Int Int -> Property
+prop_foldMap m = size m > 0 ==>
+  let xs = toList m
+  in  Foldable.foldMap (:[]) m == List.map snd xs
+
+prop_foldMapWithKey :: Map Int Int -> Property
+prop_foldMapWithKey m = size m > 0 ==>
+  let xs = toList m
+  in  foldMapWithKey (\k v -> [k,v]) m == concatMap (\(k,v) -> [k,v]) xs
+
+prop_foldr :: Int -> Map Int Int -> Property
+prop_foldr n m = size m > 0 ==>
+  let xs = toList m
   in  foldr (+) n m == List.foldr (+) n (List.map snd xs) &&
-      foldr (:) [] m == List.map snd (List.sort xs) &&
-      foldrWithKey (\_ a b -> a + b) n m == List.foldr (+) n (List.map snd xs) &&
+      foldr (:) [] m == List.map snd xs
```

Comment:
Why not just

```haskell
prop_foldr :: Fun (A, B) B -> B -> Map Int A -> Property
prop_foldr c n m = foldr c' n m = foldr c' n (F.toList m)
  where
     c' = curry (apply f)
```

Reply:
`Data.Foldable.toList` and `elems` are implemented in terms of `foldr`, so I didn't use them for it, but I did this for the rest of the folds.